server/src/main/java/io/crate/fdw/JdbcForeignDataWrapper.java

```diff
@@ -144,16 +144,15 @@ public CompletableFuture<BatchIterator<Row>> getIterator(Role currentUser,
 
         Settings options = server.options();
         String url = urlSetting.get(options);
-        URI uri;
-        try {
-            uri = new URI(url);
-        } catch (URISyntaxException e) {
-            throw new RuntimeException(e);
+
+        var parsedURL = parseURL(url, null);
```

Comment:
That would **only** work with PostgreSQL Servers as pgjdbc specifcially checks for
```java
if (!urlServer.startsWith("jdbc:postgresql:")) {
      LOGGER.log(Level.FINE, "JDBC URL must start with \"jdbc:postgresql:\" but was: {0}", url);
      return null;
    }
```

Reply:
I think stripping `jdbc:` with 

```java
uri = new URI(url.substring(5));
```

would be a more general solution, even though it won't cover special cases like:

```
jdbc:oracle:thin:@//[HOST][:PORT]/SERVICE
jdbc:mysql:loadbalance://[HOST][:PORT]/SERVICE
```

Reply:
Thanks for your input @proddata . I also tried to find a general solution but could not find a way to cover all cases so ended up with the most narrow one.

Since we [bundled only Postgres JDBC driver](https://cratedb.com/docs/crate/reference/en/master/admin/fdw.html#jdbc), the connection string must start with `jdbc:postgresql:` for now anyways. I think as we include more JDBC drivers in the future, this logic can be expanded as well(and depend on the driver for parsing the connection strings).

Reply:
While our system comes with the Postgres JDBC driver pre-packaged, it's possible to utilize additional drivers by adding them to the /lib directory. The existing approach, which effectively restricts all JDBC URLs not pertaining to PostgreSQL, may be overly cautious. As far as I understand, the initial rationale behind this limitation was to prevent regular CrateDB administrators from gaining superuser privileges. 

---

A overall better approach might be to prevent any user mapping for the superuser `crate` and have a setting for enabling the use of `crate`.

Reply:
I am fine with string manipulation if we can double check with `DriverManager.getDriver(url)` which calls `Driver.acceptsURL(url)`.